src/ckb-daemon/structures.h

```diff
@@ -247,7 +248,7 @@ typedef struct {
     // Color dithering in use
     char dither;
     // Flag to check, if large macros should be sent delayed
-    char delay;
+    uint delay;
```

Comment:
I sort of agree, option #4. Perhaps add a warning somewhere in the README about really long delays and keyboard overflows.

I have not yet tested your branch on my Mac. Keyboard is on a different machine than I've been at since last week. Hope to just double check and see if there's a crash or just a really long playback.